### ISSUE TITLE
using pandas read csv cause gibberish

### ISSUE THREAD

**Issue by JVfisher** [spyder-ide/spyder#3432]
## Description of your problem

**What steps will reproduce the problem?**

1. update spyder to 3.0.0.b7
2. use read_csv in pandas and specify encoding in read_csv to gbk
3. prepare a csv which filling with chinese character and encoded using gbk

**What is the expected output? What do you see instead?**
in 3.0.0b6, we can see the right character, but in 3.0.0b7 we just get some gibberish

**Please provide any additional information below**

## Versions and main components

* Spyder Version:3.0.0b7
* Python Version: Python 2.7.12 |Anaconda 4.0.0 (64-bit)
* Operating system: win7

## Dependencies
pandas  0.18.1
Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

**Comment by ccordoba12** [comment]
Please provide some file to test this error. Thanks :-)

**Comment by JVfisher** [comment]
here is a sample csv:

username,DIY玩具,HIFI,gold,纪录片,health,wine,book,中西药品,tools
((江苏XX))社,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海)XX有限公司,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海)XX设备科技有限公司,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海luoke设备有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海jisheng有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海金融新居服务有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海居室大有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(中国电商在线有限公司),0.0,0.0,0.0,0.0,1408.6,0.0,0.0,0.0,0.0
(vip客户）顺义开发集团,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(偶发事件探索有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(python研发硬件有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0

**Comment by JVfisher** [comment]
I don't know why I can't upload the txt or csv file. I tried several times and just got "Something went really wrong, and we can’t process that file"

**Comment by JVfisher** [comment]
here is what I got using 3.0.0b6 
![Uploading reading.PNG…]()

**Comment by jitseniesen** [comment]
I tried but failed to reproduce this on Linux. This is the script that I executed:
``` python
# coding=utf-8
contents = u"""username,DIY玩具,HIFI,gold,纪录片,health,wine,book,中西药品,tools
((江苏XX))社,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海)XX有限公司,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海)XX设备科技有限公司,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海luoke设备有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海jisheng有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海金融新居服务有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(上海居室大有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(中国电商在线有限公司),0.0,0.0,0.0,0.0,1408.6,0.0,0.0,0.0,0.0
(vip客户）顺义开发集团,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(偶发事件探索有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
(python研发硬件有限公司),0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0"""
with open('foo.csv', 'w') as f:
    f.write(contents.encode('gbk'))

import pandas as pd
df = pd.read_csv('foo.csv', encoding='gbk')
```

@JVfisher What do you mean with:

> in 3.0.0b6, we can see the right character, but in 3.0.0b7 we just get some gibberish

Do you see gibberish in the IPython console or in the Variable Explorer? Do you get problems with my script, or do you do something different? Thanks

**Comment by ccordoba12** [comment]
@jitseniesen, could you check what happens on Windows?

**Comment by jitseniesen** [comment]
I'm not sure whether this is what is meant, but I found one difference between beta6 and beta7 on Windows. It seems a different encoding is used when converting a DataFrame containing Unicode strings to a (non-Unicode) string. For example, here I print a DataFrame with a single entry containing Unicode character 209, which is capital N with a tilde:

Spyder 3.0.0 beta6:
![spy-b6](https://cloud.githubusercontent.com/assets/7941918/18779070/40cb68b2-816f-11e6-9504-1b5d2c246cfe.png)

Spyder 3.0.0 beta7:
![spy-b7](https://cloud.githubusercontent.com/assets/7941918/18779098/5d58afe4-816f-11e6-866f-a2c10956d36b.png)

Versions: Python 2.7.12 64bits, Qt 5.6.0, PyQt5 5.6, on Windows 7.

**Comment by jitseniesen** [comment]
> I'm not sure whether this is what is meant, but I found one difference between beta6 and beta7 on Windows. It seems a different encoding is used when converting a DataFrame containing Unicode strings to a (non-Unicode) string.

I think this has the same cause as issue #3428: the missing `__init__.py` file. After copying `utils/site` from a Git clone to the Anaconda directory, beta7 has the same behaviour as beta6. So my guess is that this issue will disappear with the next release of Spyder.

**Comment by ccordoba12** [comment]
Closing, as it seems this was fixed in Spyder 3.0 ;-)
